src/prefs.c

```diff
@@ -261,16 +267,22 @@ static gboolean kb_tree_view_button_press_event_cb(GtkWidget *widget, GdkEventBu
 }
 
 
-static void kb_init_tree(KbData *kbdata)
+static void kb_init_tree(KbData *kbdata, GtkWidget *kb_filter_entry)
 {
 	GtkCellRenderer *renderer;
+	GtkTreeModel *filter_model;
 	GtkTreeViewColumn *column;
 
 	kbdata->tree = GTK_TREE_VIEW(ui_lookup_widget(ui_widgets.prefs_dialog, "treeview7"));
 
 	kbdata->store = gtk_tree_store_new(KB_TREE_COLUMNS,
 		G_TYPE_STRING, G_TYPE_STRING, G_TYPE_INT, G_TYPE_BOOLEAN, G_TYPE_INT);
-	gtk_tree_view_set_model(GTK_TREE_VIEW(kbdata->tree), GTK_TREE_MODEL(kbdata->store));
+	filter_model = gtk_tree_model_filter_new(GTK_TREE_MODEL(kbdata->store), NULL);
+	gtk_tree_model_filter_set_visible_func(
+		GTK_TREE_MODEL_FILTER(filter_model), kb_tree_filter_func, kb_filter_entry, NULL);
+	/* set model to tree view */
+	gtk_tree_view_set_model(GTK_TREE_VIEW(kbdata->tree), filter_model);
+	g_object_unref(filter_model);
```

Comment:
Is it correct to unref both objects?
It seems to work this way but I'm never sure with unref'ing.

Reply:
I think so (I think I would definitely forget to unref this one).

I tried to add one more `g_object_unref(filter_model);` and it reliably crashes so, experimentally, what you did is the correct balance between refs and unrefs :-)